File: FrequencyCounter/Anagram.js
function anagram(w1,w2){
    if (w1.length !== w2.length){
        return false
    }
    const lookup = {}
    for(let i = 0; i<w1.length;i++){
        let letter = w1[i]
        lookup[letter] ? lookup[letter] += 1 : lookup[letter] = 1
    }
    for(let i = 0; i<w2.length;i++){
        let letter = w2[i]
        if(!lookup[letter]){
            return false
        }
        else {
            lookup[letter] -=1
        }
    }
    return true
}

console.log(anagram("aza","zaa"))

File: 01-FrequencyCounter/Anagram.py
#  Given the word count is equal we can then only use the second for
# loop to iterate over the second word and try to find the value in the lookup
# or else we might have to run the loop like we did in fc.py check it once

w1,w2 = "aza","zaa"

def Anagram(w1,w2):
    if (len(w1) != len(w2)):
        return False

    lookup = {}

    for i in w1:
        lookup[i] = (lookup.get(i,0))+1
    
    for letter in w2:
        if (letter not in lookup):
            return False
        else:
            lookup[letter] -= 1
    
    return True

a = Anagram(w1, w2)
print(a)